internal/appsec/remoteconfig.go

```diff
@@ -48,13 +48,13 @@ func mergeMaps[K comparable, V any](m1 map[K]V, m2 map[K]V) map[K]V {
 }
 
 // combineRCRulesUpdates updates the state of the given rulesManager with the combination of all the provided rules updates
-func combineRCRulesUpdates(r *rulesManager, updates map[string]remoteconfig.ProductUpdate) (map[string]rc.ApplyStatus, error) {
-	statuses := map[string]rc.ApplyStatus{}
+func (r *rulesManager) combineRCRulesUpdates(updates map[string]remoteconfig.ProductUpdate) (statuses map[string]rc.ApplyStatus, err error) {
```

Comment:
I'm in favour of keeping the separation of concerns here. Rules manager describes a ruleset and helps performing operations on it. I don't think it should be aware of RC updates and statuses.
We should be able to change what an RC update is without having to change rules manager code. Same way we shouldn't have to change the RC processing if rules manager changes (or gets replaced by another representation)

Also, the rules manager will most probably move to internal-go since it's starting to make more and more sense to start using it in datadog-agent as well (for example for API sec which requires combining rules fragments at startup)